File: golang/add-digits-258.go
package main

import "fmt"

func main() {
	fmt.Println(addDigits(123456789))
}

func addDigits(num int) int {
	for num > 9 {
		sum := 0
		numc := num
		for numc > 0 {
			sum += numc % 10
			numc /= 10
		}

		num = sum
	}

	return num

}

File: nodejs/add-digits-258.js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    'use strict';
    let numS = num.toString();
    while (numS.length > 1) {
        let sum = 0;
        for (let i = 0; i < numS.length; i++) {
            sum += parseInt(numS.substring(i,i+1));
        }
        numS = sum.toString();
    }
    return numS;
};

console.log(addDigits(8712387128));
